### PR TITLE
`positive()`, `negative()`, `positiveOrZero()` & `negativeOrZero()` introduced to `NumericConstraintBase`

### PR DESCRIPTION
**Pull request by DiegoKrupitza**
Previously we had to use `NumericConstraintBase#isGreaterThan(0)` to check if a given numeric value is positive. This is quite inconvenient, therefore I introduced `NumericConstraintBase#positive()`. As a counterpart to `positive()` I added `negative()` too.

### CODE REVIEW

File: src/main/java/am/ik/yavi/constraint/CharacterConstraint.java [hunk 1]
```diff
@@ -45,4 +45,14 @@
 	protected Predicate<Character> isLessThanOrEqual(Character max) {
 		return x -> x <= max;
 	}
+
+	@Override
+	protected Character zeroValue() {
+		return Character.MIN_VALUE;
```
Comment (making) [inline src/main/java/am/ik/yavi/constraint/CharacterConstraint.java:51]: `'0'` is explicit and better to read

File: src/main/java/am/ik/yavi/constraint/base/NumericConstraintBase.java [hunk 2]
```diff
@@ -18,10 +18,7 @@
 import java.util.function.Predicate;
 
 import static am.ik.yavi.core.NullAs.VALID;
-import static am.ik.yavi.core.ViolationMessage.Default.NUMERIC_GREATER_THAN;
-import static am.ik.yavi.core.ViolationMessage.Default.NUMERIC_GREATER_THAN_OR_EQUAL;
-import static am.ik.yavi.core.ViolationMessage.Default.NUMERIC_LESS_THAN;
-import static am.ik.yavi.core.ViolationMessage.Default.NUMERIC_LESS_THAN_OR_EQUAL;
+import static am.ik.yavi.core.ViolationMessage.Default.*;
```
Comment (making) [inline src/main/java/am/ik/yavi/constraint/base/NumericConstraintBase.java:21]: Can you run `mvn formatter:format` ?

**Review by making — CHANGES_REQUESTED** [review]

File: src/main/java/am/ik/yavi/constraint/base/NumericConstraintBase.java [hunk 3]
```diff
@@ -53,11 +50,27 @@ public C lessThanOrEqual(V max) {
...
 
 	protected abstract Predicate<V> isGreaterThanOrEqual(V min);
 
 	protected abstract Predicate<V> isLessThan(V max);
 
 	protected abstract Predicate<V> isLessThanOrEqual(V max);
+
+	protected abstract V zeroValue();
+
+	protected abstract V oneValue();
```
[the remark this answers is not in the document]
Comment (DiegoKrupitza) [reply src/main/java/am/ik/yavi/constraint/base/NumericConstraintBase.java:75]: I forgot to delete this. I wanted to implement further constraints but then I realised the PR would become way to big.

File: src/main/java/am/ik/yavi/constraint/CharacterConstraint.java (hunk 1 above)
Comment (DiegoKrupitza) [reply src/main/java/am/ik/yavi/constraint/CharacterConstraint.java:51]: The difference between `Character.MIN_VALUE` and '0' is that the first represent `U+0000` and the second is just the character

File: src/main/java/am/ik/yavi/constraint/base/NumericConstraintBase.java (hunk 2 above)
Comment (DiegoKrupitza) [reply src/main/java/am/ik/yavi/constraint/base/NumericConstraintBase.java:21]: Yes, forgot about that! 


Would be nice to have a contribution guideline so we know what should be done before sending a PR.

File: src/main/java/am/ik/yavi/constraint/CharacterConstraint.java (hunk 1 above)
Comment (making) [reply src/main/java/am/ik/yavi/constraint/CharacterConstraint.java:51]: Ah, right.
So is that the numeric 'zero value' ??
No tests added for `CharacterConstraintTest`.

Comment (making) [reply src/main/java/am/ik/yavi/constraint/CharacterConstraint.java:51]:
```java
System.out.println(Character.getNumericValue('0')); // => 0
System.out.println(Character.getNumericValue(Character.MIN_VALUE)); // => -1
```

So, I think `'0'` is correct.

Comment (DiegoKrupitza) [reply src/main/java/am/ik/yavi/constraint/CharacterConstraint.java:51]: Yeah as far as I can see if you cast `Character.MIN_VALUE` to integer you get the numeric value of 0

Comment (DiegoKrupitza) [reply src/main/java/am/ik/yavi/constraint/CharacterConstraint.java:51]: Ok I will change it to '0'

Comment (making) [reply src/main/java/am/ik/yavi/constraint/CharacterConstraint.java]: Never mind. `Character.MIN_VALUE` is correct.

```java
int a = '0';
System.out.println(a); // => 48
int b = Character.MIN_VALUE;
System.out.println(b); // => 0
```

I think I was confused by the `oneValues` implementation...

```java
	@Override
	protected Character oneValue() {
		return '1';
	}
```

Comment (DiegoKrupitza) [reply src/main/java/am/ik/yavi/constraint/CharacterConstraint.java]: Oh... i will change it back to `Character.MIN_VALUE`

**Review by making — APPROVED** [review]

**Comment by making** [comment]
Thanks, I'll merge the pr after bumping pom.xml to 0.10.0-SNAPSHOT

**Comment by DiegoKrupitza** [comment]
> Thanks, I'll merge the pr after bumping pom.xml to 0.10.0-SNAPSHOT

Perfect! I am looking forward to it 🤩 :shipit:

**Pull request merged**